src/pybind/mgr/cephadm/module.py

```diff
@@ -1889,18 +1889,15 @@ def _create_daemon(self, daemon_type, daemon_id, host,
                        keyring=None,
                        extra_args=None, extra_config=None,
                        reconfig=False,
-                       osd_uuid_map=None):
+                       osd_uuid_map=None,
+                       custom_config=None,  # type: Optional[Callable[[], Dict[str, Any]]]
```

Comment:
how does this relate to `extra_config`?

Reply:
`extra_config` is added to the configuration generated for cluster daemons. `custom_config` is used for generating configuration for non-cluster native daemons (currently: prometheus, grafana). `extra_config` is ignored if `custom_config` has been passed.